src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunHardwareIntrinsicRootProvider.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+using Internal.TypeSystem;
+using Internal.JitInterface;
+using System.Diagnostics;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Root all methods on supported hardware intrinsic classes.
+    /// </summary>
+    public class ReadyToRunHardwareIntrinsicRootProvider(ReadyToRunCompilerContext context) : ICompilationRootProvider
+    {
+        public void AddCompilationRoots(IRootingServiceProvider rootProvider)
+        {
+            InstructionSetSupport specifiedInstructionSet = context.InstructionSetSupport;
+            TargetArchitecture targetArch = context.Target.Architecture;
+
+            // Hardware intrinsics can only live in the system module.
+            foreach (MetadataType type in context.SystemModule.GetAllTypes())
```

Comment:
This could be filtered down considerably.

Hardware intrinsics can only live in the `System.Runtime.Intrinsics` namespace and they are setup in a very particular way, so you can only have one subnamespace reporting support.